mmdet3d/datasets/pipelines/transforms_3d.py

```diff
@@ -650,6 +650,288 @@ def __repr__(self):
+                                                     (cur_max + 0.001)),
+                    axis=1) == 3
+
+                if not cur_choice.any():  # no points in this patch
+                    continue
+
+                # sample points in this patch to multiple batches
+                cur_center = cur_min + self.block_size / 2.0
+                point_idxs = np.where(cur_choice)[0]
+                num_batch = int(
+                    np.ceil(point_idxs.shape[0] / float(self.num_points)))
+                point_size = int(num_batch * self.num_points)
+                replace = point_size > 2 * point_idxs.shape[0]
+                point_idxs_repeat = np.random.choice(
+                    point_idxs,
+                    point_size - point_idxs.shape[0],
+                    replace=replace)
+                choices = np.concatenate([point_idxs, point_idxs_repeat])
+                np.random.shuffle(choices)
+
+                # construct model input
+                point_batch = self._input_generation(coords[choices],
+                                                     cur_center, coord_max,
+                                                     attributes[choices],
+                                                     attribute_dims,
+                                                     point_type)
+
+                all_points.append(point_batch)
+                all_idxs.append(choices)
+
+        all_points = point_type.cat(all_points)
+        all_idxs = np.concatenate(all_idxs)
+
+        # make sure all points are sampled at least once
+        assert np.unique(all_idxs).shape[0] == points.shape[0]
+
+        return all_points
+
+    def __call__(self, results):
+        """Call function to sample points to in indoor scenes.
+
+        Args:
+            input_dict (dict): Result dict from loading pipeline.
+
+        Returns:
+            dict: Results after sampling, 'points', 'pts_instance_mask' \
+                and 'pts_semantic_mask' keys are updated in the result dict.
+        """
+        points = results['points']
+
+        if self.test_mode:
+            points = self._sliding_patch_generation(points)
+            results['points'] = points
+            return results
+
+        pts_semantic_mask = results['pts_semantic_mask']
+
+        points, choices = self._patch_points_sampling(points,
+                                                      pts_semantic_mask)
+
+        results['points'] = points
+        results['pts_semantic_mask'] = pts_semantic_mask[choices]
+        pts_instance_mask = results.get('pts_instance_mask', None)
+        if pts_instance_mask is not None:
+            results['pts_instance_mask'] = pts_instance_mask[choices]
+
+        return results
+
+    def __repr__(self):
+        """str: Return a string that describes the module."""
+        repr_str = self.__class__.__name__
+        repr_str += '(num_points={},'.format(self.num_points)
```

Comment:
Use f-string? I noticed that in `transforms_3d.py` all `__repr__` use `.format`, probably ask for the opinion of @ZwwWayne

Reply:
You are right, as I also find this very confusing. Some functions use ```.format``` for ```__repr__``` such as [this](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/datasets/pipelines/transforms_3d.py#L110-L116), while some others use f-string such as [this](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/datasets/pipelines/transforms_3d.py#L210-L220). This contradiction also happens in ```pipelines/loading.py```. Ask @ZwwWayne here.

Reply:
That's an issue left over from history. Just use f-string in new PRs. You can also modify some old implementations that use `.format`.

Reply:
Fix.